src/app/games/siam/tests/siam.component.spec.ts

```diff
@@ -151,15 +142,122 @@ describe('SiamComponent', () => {
+    }));
+    it('should select the other piece when player selects another piece instead of a target', fakeAsync(async() => {
+        // Given a state with a piece, and the player being in the middle of creating a move
+        const board: Table<SiamPiece> = [
+            [_, _, _, _, _],
+            [_, _, _, _, _],
+            [_, M, M, M, _],
+            [_, _, _, U, _],
+            [_, _, _, _, _],
+        ];
+        const state: SiamState = new SiamState(board, 0);
+        testUtils.setupState(state);
+        await testUtils.expectClickSuccess('#remainingPieces_0');
+
+        // When the player clicks on the piece on the board
+        await testUtils.expectClickSuccess('#square_3_3');
+
+        // Then a new move should be in creation and the player can finish the move
+        await testUtils.expectClickSuccess('#square_3_4');
+        const move: SiamMove = SiamMove.of(3, 3, MGPOptional.of(Orthogonal.DOWN), Orthogonal.DOWN).get();
+        await testUtils.expectMoveSuccess('#orientation_DOWN', move);
+    }));
+    it('should cancel the move attempt when clicking on invalid target instead of selecting orientation move', fakeAsync(async() => {
+        // Given a state with a piece already on board, which has been selected by the player, along with a valid target
+        const board: Table<SiamPiece> = [
+            [_, _, _, _, _],
+            [_, _, _, _, _],
+            [_, M, M, M, _],
+            [_, _, _, _, _],
+            [_, _, _, _, U],
+        ];
+        const state: SiamState = new SiamState(board, 0);
+        testUtils.setupState(state);
+        await testUtils.expectClickSuccess('#square_4_4');
+        await testUtils.expectClickSuccess('#square_4_3');
+
+        // When the player clicks on a square instead of an orientation arrow
+        spyOn(testUtils.getComponent(), 'cancelMoveAttempt').and.callThrough();
+        await testUtils.expectClickSuccess('#square_2_2');
+        // Then the move should be cancelled
+        expect(testUtils.getComponent().cancelMoveAttempt).toHaveBeenCalledOnceWith();
+    }));
+    it('should cancel the move when clicking on invalid target for move', fakeAsync(async() => {
+        // Given a state with a piece already on board, which has been selected by the player
+        const board: Table<SiamPiece> = [
+            [_, _, _, _, _],
+            [_, _, _, _, _],
+            [_, M, M, M, _],
+            [_, _, _, _, _],
+            [_, _, _, _, U],
+        ];
+        const state: SiamState = new SiamState(board, 0);
+        testUtils.setupState(state);
+        await testUtils.expectClickSuccess('#square_4_4');
+
+        // When the player clicks on an empty square which is not a valid target
+        // Then the move should be cancelled and an error should be toasted
+        await testUtils.expectClickFailure('#square_2_2', RulesFailure.MUST_CHOOSE_PLAYER_PIECE());
```

Comment:
Et la typiquement, non, il devait pas sélectionner une pièce du joueur, là faudrait dire "should click on neighboring space to chose landing space" ?

Reply:
Je suis d'avis qu'un clic = un changement à l'UI, que ce soit un toast, une progression dans la construction du mouvement, ou une annulation de mouvement. Donc clic sur case vide qui annule le mouvement ça irait non ? Clic sur une montagne pourrait annuler le mouvement simplement, sans toaster également. De plus le joueur est déjà informé des clics valides avec les highlights

Reply:
Oui, c'est peut être très bien mais j'ai l'impression qu'on dit plus rien à l'utilisateur, et que même si c'est plus clair, y a un petit côté "démerde toi!" :D 
Mais oui, en effet, un clic invalide quand t'as pas ta pièce sélectionnée devrais dire "must choose one of your piece"
Puis un clic invalide au lieu de choisir une valid landing case devrais dire "votre pièce sélectionée ne peut pas atterir là, choisissez une des cases highlightées"
Puis un clic invalide au lieu de choisir l'orientation devrais dire "AHLOOOO CONAR ? Tu vois pas les grandes flèches là ? QUICHE LORRAINE"
Là on a que le premier j'crois.

Reply:
(ps: still describing la montagne as an empty case here!!!)

Reply:
ok donc le tout est fait:
  - [given: aucun mouvement en cours de construction] clic sur une case sans une de tes pièces -> tu dois choisir un pièce monsieur (déjà le cas avant)
  - [given: pièce choisie] clic sur une destination non valide (excepté une autre pièce) -> tu dois choisir une destination valide
  - [given: pièce + destination choisie] clic anywhere -> tu dois choisir une sélection

(TODO: tester)